Loop/Managers/LoopAppManager.swift

```diff
@@ -297,9 +305,47 @@ extension LoopAppManager: AlertPresenter {
         rootViewController?.topmostViewController.present(viewControllerToPresent, animated: animated, completion: completion)
     }
 
-    func dismiss(animated: Bool, completion: (() -> Void)?) {
+    func dismissTopMost(animated: Bool, completion: (() -> Void)?) {
         rootViewController?.topmostViewController.dismiss(animated: animated, completion: completion)
     }
+
+    func dismissAlert(_ alertToDismiss: UIAlertController, animated: Bool, completion: (() -> Void)?) {
+        if rootViewController?.topmostViewController == alertToDismiss {
+            dismissTopMost(animated: animated, completion: completion)
+        } else {
+            // check if the alert to dismiss is presenting another alert (and so on)
+            // calling dismiss() on an alert presenting another alert will only dismiss the presented alert
+            // (and any other alerts presented by the presented alert)
+
+            // get the stack of presented alerts that would be undesirably dismissed
+            var presentedAlerts: [UIAlertController] = []
+            var currentAlert = alertToDismiss
+            while let presentedAlert = currentAlert.presentedViewController as? UIAlertController {
+                presentedAlerts.append(presentedAlert)
+                currentAlert = presentedAlert
+            }
+
+            if presentedAlerts.isEmpty {
+                alertToDismiss.dismiss(animated: animated, completion: completion)
+            } else {
+                // Do not animate any of these view transitions, since the alert to dismiss is not at the top of the stack
+                // dismiss all the child presented alerts
+                alertToDismiss.dismiss(animated: false) {
```

Comment:
I don't understand why calling `alertToDismiss.dismiss` is the right thing to do here? Shouldn't this be dismissing all of the `presentedAlerts`, one at a time?

Reply:
from [apple docs](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621505-dismiss) on `dismiss()`:

> If you present several view controllers in succession, thus building a stack of presented view controllers, calling this method on a view controller lower in the stack dismisses its immediate child view controller and all view controllers above that child on the stack.

This in fact was the root issue for this bug. `dismiss()` was being called on the `UIAlertController` thinking it would dismiss that `UIAlertController` but instead it was dismissing the child presented `UIAlertController`. For the Dexcom expire soon alerts, they transition from one to the next in the same communication interval. Thus an async dismiss and present are called back-to-back and the order is determined by the Dexcom SDK.

Reply:
I'll update the comment to help clarify.